port/unix/omrsock.c

```diff
@@ -361,7 +362,20 @@ omrsock_freeaddrinfo(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t hand
 int32_t
 omrsock_sockaddr_init(struct OMRPortLibrary *portLibrary, int16_t family, char *addr, uint16_t port, omrsock_sockaddr_t handle)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	omr_os_sockaddr_in *sockAddr = NULL;
+	int32_t osFamily = 0;
+
+	sockAddr = (omr_os_sockaddr_in *) &handle->data;
+	memset(handle, 0, sizeof(struct OMRSockAddrStorage));
+	osFamily = get_os_family(family);
+
+	sockAddr->sin_family = osFamily;
+	sockAddr->sin_port = htons(port);
+	if (1 != inet_pton(osFamily, addr, &sockAddr->sin_addr)) {
+		return OMRPORT_ERROR_SOCK_INET_PTON_FAILED;
+	}
+
+	return 0;	
```

Comment:
I think this can be simplified:
```c++
	memset(handle, 0, sizeof(struct OMRSockAddrStorage));

	handle->data->sin_family = get_os_family(family);
	handle->data->sin_port = htons(port); // TODO: remove call to htons 

	if (1 != inet_pton(osFamily, addr, &handle->data->sin_addr)) {
		return OMRPORT_ERROR_SOCK_INET_PTON_FAILED;
	}

	return 0;
```

Reply:
Yup, sounds good. I will fix this.